src/main/java/org/apache/mesos/hdfs/state/LiveState.java

```diff
@@ -21,8 +21,9 @@
   public static final Log log = LogFactory.getLog(LiveState.class);
   private Set<Protos.TaskID> stagingTasks = new HashSet<>();
   private AcquisitionPhase currentAcquisitionPhase = AcquisitionPhase.RECONCILING_TASKS;
-  // TODO (nicgrayson) Might need to split this out to jns, nns, and dns if dns too big
```

Comment:
Since you're not fixing getJournalNodeSize()/getNameNodeSize() to return the number of running JNs/NNs without iterating (filtering) through all runningTasks, I don't think you can really remove this TODO. Your new HashMaps cover staging and running tasks, so they won't be able to do that job. We may still want to have other separate lists of /running/ JNs and NNs.